WDL/runtime/cache.py

```diff
@@ -44,15 +62,34 @@ def get(
         opens shared flocks on all files referenced therein, which will remain for the life of the
         CallCache object.
         """
-        raise NotImplementedError()
+        from .. import values_from_json
+
+        file_path = os.path.join(self.outputs_cache_dir, f"{key}.json")
+
+        try:
+            with open(file_path, "rb") as file_reader:
+                contents = file_reader.read()
+        except FileNotFoundError:
+            return None
+        contents = json.loads(contents)
+        return values_from_json(contents, output_types)
 
     def put(
-        self, key: str, outputs: Env.Bindings[Value.Base], logger: Optional[logging.Logger] = None
+        self,
+        key: str,
+        run_dir: str,
+        outputs: Env.Bindings[Value.Base],
+        logger: Optional[logging.Logger] = None,
     ) -> None:
         """
         Store call outputs for future reuse
         """
-        raise NotImplementedError()
+        from .. import values_to_json
+
+        with open(os.path.join(self.outputs_cache_dir, f"{key}.json"), "w+") as outfile:
```

Comment:
We have a utility function `write_atomic` you can add to the `from .._util import ...`, which handles writing out a string to a temp file, then renaming it to the desired filename (usually an atomic op). It prevents the possibility that another thread/process opens the file at the same instant we're writing it and sees an incomplete value. (SQLite would also address such issues)

Reply:
👍